client/src/components/TabPane/Cards/MessagesCard.js

```diff
@@ -1,7 +1,462 @@
+    },
+    {
+      title: "Tags",
+      dataIndex: "tags",
+      editable: true,
+    },
+    {
+      title: "Operations",
+      render: (record) => {
+        return (
+          <>
+            <EditOutlined
+              onClick={() => {
+                handleEdit(record);
+              }}
+            />
+            <DeleteOutlined
+              onClick={() => {
+                handleDelete(record);
+              }}
+              style={{ color: "red", marginLeft: 12 }}
+            />
+          </>
+        );
+      },
+    },
+  ];
+
+  const deleteMessage = (id) => {
+    fetch(`${baseURL}/message/${id}`, {
+      method: "DELETE",
+    })
+      .then((response) => {
+        fetchData();
+        if (response.ok) return response;
+        else return response.json();
+      })
+      .then((res) => {
+        if (res.ok) {
+          success("Message deleted successfully!");
+        } else {
+          error(res.message);
+        }
+      })
+      .catch((err) => {
+        console.log("Error while deleting: ", err);
+        error("Error! Couldn't delete message");
+      });
+  };
+
+  const saveEdited = () => {
+    fetch(`${baseURL}/message`, {
+      method: "PUT",
+      headers: {
+        Accept: "application/json, text/plain",
+        "Content-Type": "application/json;charset=UTF-8",
+      },
+      body: JSON.stringify(editingMessage),
+    })
+      .then((response) => {
+        fetchData();
+        return response.json();
+      })
+      .then((res) => {
+        if (res.id) {
+          success("Message edited successfully!");
+          resetEditing();
+        } else {
+          error(res.message);
+        }
+      })
+      .catch((err) => {
+        console.log("Error while edditing message: ", err);
+        error("Error! Couldn't edit message");
+      });
+  };
+
+  const saveNewMessage = () => {
+    fetch(`${baseURL}/message`, {
+      method: "POST",
+      headers: {
+        Accept: "application/json, text/plain",
+        "Content-Type": "application/json;charset=UTF-8",
+      },
+      body: JSON.stringify(addingMessage),
+    })
+      .then((response) => {
+        fetchData();
+        return response.json();
+      })
+      .then((res) => {
+        if (res.id) {
+          success("New message added successfully!");
+          resetAdding();
+        } else {
+          error(res.message);
+        }
+      })
+      .catch((err) => {
+        console.log("Error while adding new message: ", err);
+        error("Error! Couldn't add message");
+      });
+  };
+
+  const fetchLanguages = () => {
+    fetch(`${baseURL}/language`)
+      .then((response) => response.json())
+      .then((data) => {
+        for (let i = 0; i < data.length; i++) {
+          data[i]['label'] = data[i]['language'];
+          data[i]['value'] = data[i]['language'];
+        }
+        if (data.length === 0)
+        {
+          data.push({'label': 'English', 'value': 'English'});
```

Comment:
I'm not sure what are you trying to do here and if it really works in the app

Reply:
It's because we need an English language to create anything and it's not available by default in the database

Reply:
Ah, removed at the end, it's better to add it by default in the database.